ir/src/lib.rs

```diff
@@ -46,6 +46,7 @@ const NEXT_ROW: usize = 1;
 pub struct AirIR {
     air_name: String,
     constants: Constants,
+    segment_width: Vec<u16>,
```

Comment:
nit: I think I'd prefer `segment_widths` instead, to be clearer that it has the width of all the segments.